examples/kalman_filter.ipynb

```diff
@@ -0,0 +1,490 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Kalman Filter"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This example optimizes the parameters of a [Kalman-Filter](https://en.wikipedia.org/wiki/Kalman_filter).\n",
+    "\n",
+    "This example is available as a Jupyter notebook [here](https://github.com/patrick-kidger/diffrax/blob/main/examples/kalman_filter.ipynb)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import functools as ft\n",
+    "from types import SimpleNamespace\n",
+    "from typing import Optional\n",
+    "\n",
+    "import diffrax as dfx\n",
+    "import equinox as eqx  # https://github.com/patrick-kidger/equinox\n",
+    "import jax.numpy as jnp\n",
+    "import jax.random as jr\n",
+    "import jax.tree_util as jtu\n",
+    "import matplotlib.pyplot as plt\n",
+    "import optax  # https://github.com/deepmind/optax"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We use [Equinox](https://github.com/patrick-kidger/equinox) to build the Kalman-Filter implementation and to represent linear, time-invariant systems (LTI systems). \n",
+    "\n",
+    "We use [Optax](https://github.com/deepmind/optax) for optimisers (Adam etc.)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Problem Formulation\n",
+    "\n",
+    "Assume that there exists some *unknown* dynamical system of the form\n",
+    "\n",
+    "$\\frac{dx}{dt} = f(x(t), u(t), t)$\n",
+    "\n",
+    "$y(t) = g(x(t)) + \\epsilon$ \n",
```

Comment:
Nit: we could make the time dependence explicit with `\frac{dx}{dt}(t)` and `\epsilon(t)`.

Reply:
Sure